ui/components/pagination.js

```diff
@@ -0,0 +1,95 @@
+import { useRouter } from 'next/router'
+import { ChevronLeftIcon, ChevronRightIcon } from '@heroicons/react/solid'
+
+function getPageNums(selected, count, totalPages) {
+  const pageNums = []
+  const beginOffset = Math.max(3, 6 + selected - totalPages)
+  for (
+    let i = Math.max(1, selected - beginOffset);
+    pageNums.length <= count && i <= totalPages;
+    i++
+  ) {
+    pageNums.push(i)
+  }
+
+  return pageNums
+}
+
+function Pages({ path, selected, count, totalPages }) {
+  return getPageNums(selected, count, totalPages).map(page => (
+    <a
+      href={path + '?p=' + page}
+      className={`inline-flex items-center py-2 px-4 text-sm font-medium text-gray-500 hover:text-violet-300 ${
+        selected === page ? 'rounded-md bg-gray-700' : ''
+      }`}
+      key={page}
+    >
+      {page}
+    </a>
+  ))
+}
+
+function LeftArrow({ path, page }) {
```

Comment:
can we try to put `LeftArrow` and `RightArrow` to the same component?
So it would be something like this 

``function Arrow({path, direction}) {return (<a
      href={path}
      className=' inline-flex items-center pr-1 text-sm font-medium text-gray-500 hover:text-violet-300 '
    >{direction === 'left' && <ChvronLeftIcon ... />}{direction === 'RIGHT' && <ChvronRightIcon ... />}</a>)}``

Reply:
And we should probably calculate the `path` outside of this component - so the component should be a dummy where it doesn't have any logic but just render and display what it is being told to